core/arch/arm/pta/system.c

```diff
@@ -40,6 +52,66 @@ static TEE_Result system_rng_reseed(struct tee_ta_session *s __unused,
 	return TEE_SUCCESS;
 }
 
+static TEE_Result system_derive_ta_unique_key(
+			struct tee_ta_session *s,
+			uint32_t param_types,
+			TEE_Param params[TEE_NUM_PARAMS])
+{
+	size_t data_len = sizeof(TEE_UUID);
+	TEE_Result res = TEE_ERROR_GENERIC;
+	uint8_t *data = NULL;
+	uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+					  TEE_PARAM_TYPE_MEMREF_OUTPUT,
+					  TEE_PARAM_TYPE_NONE,
+					  TEE_PARAM_TYPE_NONE);
+
+	if (exp_pt != param_types ||
+	    params[0].memref.size > TA_DERIVED_EXTRA_DATA_MAX_SIZE ||
+	    params[1].memref.size < TA_DERIVED_KEY_MIN_SIZE ||
+	    params[1].memref.size > TA_DERIVED_KEY_MAX_SIZE)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	 /* The derived key shall not end up in non-secure memory by mistake */
+	if (!tee_vbuf_is_sec(params[1].memref.buffer, params[1].memref.size)) {
+		res = TEE_ERROR_SECURITY;
+		goto err;
+	}
+
+	/*
+	 * Check for user provided data that could be mixed together with the
+	 * TA UUID.
+	 */
+	if (params[0].memref.size > 0)
+		if (ADD_OVERFLOW(data_len, params[0].memref.size, &data_len)) {
+			res = TEE_ERROR_SECURITY;
+			goto err;
+		}
+
+	data = calloc(data_len, sizeof(*data));
```

Comment:
`sizeof(*data)` seems like a complicated way of writing `1`.

Reply:
Yeps, the type for data will probably never change, so `1` will do.